### DISCUSSION TITLE
Display Order Options

### DISCUSSION THREAD

**Discussion by djmega152** [mealie-recipes/mealie#2874] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

When you add a lot of recipes, the display can become overwhelming.  Yes you can use categories and tags to filter but the recipes still display according to the time at which they were created.  

### Please provide a concise description of the feature that would resolve your issue.

It would be helpful if you were able to set options for grouping your recipes according to category or tag.  It would also be nice to be able to define the sort order of the groupings.  Think about it this way, say I am hosting a gathering and I want to look at all of my recipes for that gathering in one easy place.  Yes I can create a meal planner or a cookbook but within those views, it's a flat list.  With this idea I would like to say first show me the items with a category of "Apps" then show me "BBQ" followed by "Sides" and finally "Dessert"  These sections would display the recipes associated with those sections.  Having filters that can be defined by section (similar to cookbooks) would increase the power of this feature.  

### Please consider and list out some caveats or tradeoffs made in your design decision

I can't think of any caveats other than the challenge of filters within filters if that's how this is done.  If you create a cookbook with filters and then your sections have filters, how do you handle conflicts in the filter criteria.  Something to consider.  

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [X] I'm willing to sponsor/pay a developer to do this work

**Comment by ChipSkylark37** [comment]
I've created a new Feature Request for a more advanced searching / filtering capability, and think this is related: https://github.com/mealie-recipes/mealie/discussions/4808

For example, selecting these two categories:

![image](https://github.com/user-attachments/assets/8b80c6d0-1d26-445d-acdb-f0718746ba19)

Basically just shows you the full list of both, without telling you which one is which:

![image](https://github.com/user-attachments/assets/6d5d176d-23c7-40ff-ba74-d5f17b6704d8)

If the 'Tag' was set on them, you would see that, but the Category doesn't show here.

I would have thought that the 'Toggle View' would do something similar to what you are saying, but it really only changes the scale of the Recipe IU (Large <> Small).

Instead, I would think that button should instead be similar to Windows where you can have different selectors:

![image](https://github.com/user-attachments/assets/0a4f46ab-ac4c-4dff-b0c6-858ad5292697)

There is also another selection window (maybe incorporated into this same menu?) that goes over grouping:

![image](https://github.com/user-attachments/assets/cba1bf3f-f153-41a5-8be1-193f9903790e)

Then you could see them in a Category View (Starting with App, Entree, Side, Dessert, etc) and be able to better manage what you want to make, add to a meal planner, etc.
